scripts/linters/python_linter.py

```diff
@@ -117,6 +117,44 @@ def _check_import_order(self):
                 summary_messages.append(summary_message)
         return summary_messages
 
+    def _check_non_test_files(self):
+        """This function is used to check that function
+           with test_only in their names are in test files.
+        """
+        if self.verbose_mode_enabled:
+            python_utils.PRINT('Starting function defintion checks')
+            python_utils.PRINT('----------------------------------------')
+        summary_messages = []
+        files_to_check = self.py_filepaths
+        with linter_utils.redirect_stdout(sys.stdout):
+            failed = False
+            for filepath in files_to_check:
+                if not filepath.endswith('_test.py'):
+                    file_content = FILE_CACHE.read(filepath)
+                    for line_number, line in enumerate(
+                            file_content.split('\n')):
+                        ind1 = line.find(' def ')
+                        ind2 = line.find('test_only')
```

Comment:
Find a way to use `startswith` here. maybe use `split()` to get two different strings

Reply:
yes that can work , but I don't think the existing code will create any problem, since i am searching for " def " with spaces on both side of (def) which means only lines having def with spaces at either sides  will be encountered and that will account only function definitions since def is a keyword.

Reply:
test_only will always be used at the beginning of the function name, so we must need startswith

Reply:
done